### PR TITLE
Modernize syntax

### PR DESCRIPTION
**Pull request by haampie**
Just use the Julia 0.6 syntax everywhere.

One bug fix in this PR: the stationary solvers used `SingularError` which is probably removed long ago. I'm now using `Base.LinAlg.SingularException` and added a test for it.

### CODE REVIEW

File: src/history.jl [hunk 1]
```diff
@@ -256,8 +256,8 @@ nrests(ch::RestartedHistory) = Int(ceil(ch.iters/ch.restart))
 Determine whether a collection `x` is plotable. Only vectors and matrices are
 such objects.
 """
-plotable{T<:Real}(::VecOrMat{T})::Bool = true
```
Comment (andreasnoack) [inline src/history.jl]: Is it on purpose that the return type annotations are removed here?

File: src/stationary.jl [hunk 2]
```diff
@@ -41,7 +43,7 @@ function jacobi_method!(log::ConvergenceHistory, x, A::AbstractMatrix, b;
 			for j=[1:i-1;i+1:n]
 				xi += A[i,j]*xold[j]
 			end
-			A[i,i]==0 && throw(SingularError())
+			A[i,i]==0 && throw(SingularException(1))
```
Comment (andreasnoack) [inline src/stationary.jl]: Maybe `SingularException(i)`

File: src/history.jl (hunk 1 above)
Comment (haampie) [inline src/history.jl]: To be honest I didn't know how to combine `where` syntax with return type annotations. But I don't think we need explicit return type annotations for such a simple function anyways.

Comment (andreasnoack) [inline src/history.jl]: Just asked and it seems to be a bug when using the "assignment form" of function definitions. I agree that it should really matter here.

**Pull request merged**